Documentation: utf8proc.jl tests

randyzwitch opened the pull request:
Work towards #23392 

- normalize_string a bit weak, looked for a test suite quickly but
didn’t find
- graphemes seems to return iterators of letters, not sure how to test

randyzwitch committed:
utf8proc.jl tests

- normalize_string a bit weak, looked for a test suite quickly but
didn’t find
- graphemes seems to return iterators of letters, not sure how to test

fredrikekre committed:
add spaces, move a misplaced triple backtick

fredrikekre commented on base/strings/utf8proc.jl:
````diff
@@ -28,6 +28,15 @@ isvalid(value)
+
+julia> isvalid(Char(0xd799))
+true
+```
````
Actually, should these examples be on the method just above?

randyzwitch replied:
Corrected

KristofferC commented on base/strings/utf8proc.jl:
````diff
@@ -431,13 +470,37 @@ ispunct(c::Char) = (UTF8PROC_CATEGORY_PC <= category_code(c) <= UTF8PROC_CATEGOR
+
+# Examples
+```jldoctest
+julia> isspace('\n')
````
Perhaps something like

```
julia> isspace.(['\n', '\r', ' ', '\x20'])
4-element BitArray{1}:
 true
 true
 true
 true
```

saves some space. But it might also just complicate things :)

randyzwitch replied:
`islower` and `isupper` had 3 each before I got to the file. We could always remove one of my examples to match, if 3 is desirable.

randyzwitch committed:
Move examples to correct method

Move example to upper method, then make additional examples for isvalid method where type specified as first argument

fredrikekre commented:
I wonder what is up with
```
Error in testset sparse/sparse:
Error During Test
  Test threw an exception of type MethodError
  Expression: !(Base.USE_GPL_LIBS) || abs(det(factorize(Symmetric(A)))) ≈ abs(det(factorize(Array(A))))
  MethodError: no method matching logabsdet(::Base.LinAlg.LDLt{Float64,SymTridiagonal{Float64,Array{Float64,1}}})
  Closest candidates are:
    logabsdet(!Matched::Base.LinAlg.UnitUpperTriangular{T,S} where S<:(AbstractArray{T,2} where T)) where T at linalg\triangular.jl:2193
    logabsdet(!Matched::Base.LinAlg.UnitLowerTriangular{T,S} where S<:(AbstractArray{T,2} where T)) where T at linalg\triangular.jl:2194
    logabsdet(!Matched::Union{LowerTriangular{T,S} where S<:(AbstractArray{T,2} where T), UpperTriangular{T,S} where S<:(AbstractArray{T,2} where T)}) where T at linalg\triangular.jl:2198
```
on both appveyor workers. Presumably unrelated but wonder what it is.
